### ISSUE TITLE
Install PipeWire

### ISSUE THREAD

**Issue by LukeShortCloud** [LukeShortCloud/winesapOS#172]
Instead of PulseAudio. PipeWire is now stable.

https://wiki.archlinux.org/title/PipeWire

**Comment by LukeShortCloud** [comment]
We need to ensure backwards compatibility packages are also installed for Alsa and PulseAudio.

The sound also needs to be muted as it can be extremely loud on the latest Intel Macs. See: https://github.com/LukeShortCloud/winesapOS/issues/77

**Comment by LukeShortCloud** [comment]
Manual install:

```
$ sudo pacman -S pipewire pipwire-media-session wireplumber
```

Backwards compatibility packages:

```
$ sudo pacman -S pipewire-alsa pipewire-jack pipewire-pulse
```

**Comment by LukeShortCloud** [comment]
Start the pipewire and compatibility service:

```
$ systemctl --user enable --now pipewire.service
$ systemctl --user enable --now pipwire-pulse.service
```

**Comment by LukeShortCloud** [comment]
https://wiki.archlinux.org/title/PipeWire

**Comment by LukeShortCloud** [comment]
Using `systemctl enable --user` does not work in a chroot. It fails with this error:

```
$ sudo arch-chroot /mnt sudo -u winesap systemctl enable --user pipewire-pulse.service
Failed to connect to bus: No medium found
```

As a workaround, we can manually create a symlink at `/mnt/home/winesap/.config/systemd/user/default.target.wants/pipewire-pulse.service` (the path will need to be created) that points to `/usr/lib/systemd/user/pipewire-pulse.service`. The same goes for the `pipewire.service` unit file.

**Comment by LukeShortCloud** [comment]
We now have PipeWire working correctly in the `pipewire` branch! The only problem left is that the audio is NOT set to 0% / mute. This can cause problems on newer Macs which are extremely loud by default.

**Comment by LukeShortCloud** [comment]
The problem is that PipeWire's backwards compatibilty with PusleAudio is only with the client (not the daemon). This means that our configuration file in `/home/winesap/.config/pulse/default.pa` no longer works.

As a workaround, we can use a user systemd unit file. There appears to be a race condition where it fails if it starts too early.

For KDE Plasma, I found through testing that a sleep of at least 0.3 is needed to workaround that race condition. The sleep cannot be longer than 3 seconds or else the startup sound will play.

```
[Unit]
Description=Mute the speakers.
After=pipewire.service pipewire-pulse.service pipewire-media-session.service graphical.target

[Service]
Type=oneshot
ExecStartPre=/bin/sleep 0.5
ExecStart=/usr/bin/pactl set-sink-volume 0 0

[Install]
WantedBy=default.target
```

Even better, I reworked the systemd unit to keep restarting until it works. This will ensure it starts soon enough. We cannot predict how long services will start on different hardware configurations.

```
[Unit]
Description=Mute the speakers.
After=pipewire.service pipewire-pulse.service pipewire-media-session.service graphical.target

[Service]
Type=oneshot
ExecStart=/usr/bin/pactl set-sink-volume 0 0
Restart=on-failure
RestartSec=0.1s

[Install]
WantedBy=default.target
```

The systemd unit should be created at `/home/winesap/.config/systemd/user/winesapos-mute.service` with a symlink to enable it at `/home/winesap/.config/systemd/user/default.target.wants/winesapos-mute.service`.

KDE Plasma disables its startup sound by default. However I enabled it for testing by going to System Settings > Personalization > Notifications > Applications > Configure... > System Services > Plasma Workspace > Login > Play a sound > OK. Other desktop environments, such as Cinnamon have a startup sound enabled by default.
